src/Uno.UI/UI/Xaml/Media/VisualTreeHelper.cs

```diff
@@ -491,9 +492,25 @@ internal static IReadOnlyList<_View> ClearChildren(UIElement view)
 
 			// The maximum region where the current element and its children might draw themselves
 			// This is expressed in the window (absolute) coordinate space.
-			var clippingBounds = element.Visual.ViewBox?.GetRect() is { } rect
-				? transformToElement.Transform(rect)
-				: Rect.Infinite;
+			Rect clippingBounds;
+			var viewBox = element.Visual.ViewBox;
+			if (viewBox is not null)
+			{
+				var viewBoxRect = viewBox.GetRect();
+				if (viewBox.IsAncestorClip)
+				{
+					var totalOffset = element.Visual.GetTotalOffset();
+					viewBoxRect.X -= totalOffset.X;
+					viewBoxRect.Y -= totalOffset.Y;
+				}
+
+				clippingBounds = transformToElement.Transform(viewBoxRect);
+			}
+			else
+			{
+				clippingBounds = Rect.Infinite;
+			}
```

Comment:
To centralize usages of the `IsAncestorClip` should we move all of this directly in `ShapeVisual` instead?